unic/ucd/normal/src/decomposition_type.rs

```diff
@@ -12,228 +12,180 @@
+        Font {
+            abbr => Font,
+            long => Font,
+            human => "Font Variant",
+        }
 
+        /// Vulgar fraction form
+        Fraction {
+            abbr => Fra,
+            long => Fraction,
+            human => "Vulgar Fraction",
+        }
 
-impl PartialCharProperty for DecompositionType {
-    fn of(ch: char) -> Option<Self> {
-        Self::of(ch)
-    }
-}
+        /// Initial presentation form (Arabic)
+        Initial {
+            abbr => Init,
+            long => Initial,
+            human => "Arabic Initial",
+        }
 
+        /// Isolated presentation form (Arabic)
+        Isolated {
+            abbr => Iso,
+            long => Isolated,
+            human => "Arabic Isolated",
+        }
 
-impl EnumeratedCharProperty for DecompositionType {
-    fn all_values() -> &'static [Self] {
-        Self::all_values()
-    }
+        /// Medial presentation form (Arabic)
+        Medial {
+            abbr => Med,
+            long => Medial,
+            human => "Arabic Medial",
+        }
 
-    fn abbr_name(&self) -> &'static str {
-        self.abbr_name()
-    }
+        /// Narrow (or hankaku) compatibility character
+        Narrow {
+            abbr => Nar,
+            long => Narrow,
+            human => "Narrow",
+        }
 
-    fn long_name(&self) -> &'static str {
-        self.long_name()
-    }
+        /// No-break version of a space or hyphen
+        NoBreak {
+            abbr => Nb,
+            long => Nobreak,
+            human => "No-Break",
+        }
 
-    fn human_name(&self) -> &'static str {
-        self.human_name()
-    }
-}
+        // TODO: Is this used?
+        #[allow(missing_docs)]
+        None {
+            abbr => None,
+            long => None,
+            human => "None",
+        }
 
+        /// Small variant form (CNS compatibility)
+        Small {
+            abbr => Sml,
+            long => Small,
+            human => "Small",
+        }
 
-pub mod abbr_names {
-    pub use DecompositionType::Canonical as Can;
-    pub use DecompositionType::Compat as Com;
-    pub use DecompositionType::Circle as Enc;
-    pub use DecompositionType::Final as Fin;
-    pub use DecompositionType::Font;
-    pub use DecompositionType::Fraction as Fra;
-    pub use DecompositionType::Initial as Init;
-    pub use DecompositionType::Isolated as Iso;
-    pub use DecompositionType::Medial as Med;
-    pub use DecompositionType::Narrow as Nar;
-    pub use DecompositionType::Nobreak as Nb;
-    pub use DecompositionType::None;
-    pub use DecompositionType::Small as Sml;
-    pub use DecompositionType::Square as Sqr;
-    pub use DecompositionType::Sub;
-    pub use DecompositionType::Super as Sup;
-    pub use DecompositionType::Vertical as Vert;
-    pub use DecompositionType::Wide;
-}
+        /// CJK squared font variant
+        Square {
+            abbr => Sqr,
+            long => Square,
+            human => "Squared",
```

Comment:
I believe we can expand all the `human` lines here to be more descriptive. For example, "Squared" here is not actually that useful. The point is that it's a "CJK squared font variant".

I think it's safe to go with the full text of the *Description* column in the table. Maybe only drop the *form* words and the parenthesis, which are more of a side note.

We can update this in a separate PR, but noticed some changes from the basic words and this item specifically caught my eye.